Sources/ClientRuntime/Networking/Http/HttpResponse.swift

```diff
@@ -19,7 +20,20 @@ public actor HttpResponse: HttpUrlResponse {
     }
 
     public func setStatusCode(newStatusCode: HttpStatusCode) {
+        let codeBeforeUpdate = self.statusCode.rawValue
         self.statusCode = newStatusCode
+        if newStatusCode.rawValue >= 200 && codeBeforeUpdate < 200 {
+            self.continuation?.resume()
+        }
+    }
+
+    public func waitForFinalStatusCode() async {
```

Comment:
Would it be useful to actually return the status code once a code 200 or higher is received?

Reply:
Changed as suggested.